lib/internal/observable.js

```diff
@@ -0,0 +1,898 @@
+          } catch (err) {
+            throw error;
+            // The spec expects us to use reportError();
+          }
+        };
+      }
+      if (complete != null) {
+        internalObserver.complete = () => {
+          try {
+            complete();
+          } catch (err) {
+            throw error;
+            // The spec expects us to use reportError();
+          }
+        };
+      }
+    }
+    const subscriber = createSubscriber(internalObserver, options);
+    if (subscriber.signal.aborted) {
+      subscriber[kClose]();
+    } else {
+      subscriber.signal.addEventListener('abort', () => {
+        try {
+          subscriber[kTeardown]();
+        } catch (err) {
+          subscriber.error(err);
+        }
+      });
+    }
+    try {
+      this.#subscriberCallback(subscriber);
+    } catch (err) {
+      subscriber.error(err);
+    }
+  }
+
+  /**
+   * @param {*} notifier
+   * @returns {Observable}
+   */
+  takeUntil(notifier) {
+    return new Observable((subscriber) => {
+      const notifierObservable = Observable.from(notifier);
+      notifierObservable.subscribe({
+        next() {
+          subscriber.complete();
+        },
+        error(error) {
+          subscriber.error(error);
+        },
+      }, {
+        signal: subscriber.signal,
+      });
+      if (!subscriber.active) return;
+      this.subscribe({
+        next(value) {
+          subscriber.next(value);
+        },
+        error(error) {
+          subscriber.error(error);
+        },
+        complete() {
+          subscriber.complete();
+        },
+      }, {
+        signal: subscriber.signal,
+      });
+    });
+  }
+
+  /**
+   * @param {Mapper} mapper
+   * @returns {Observable}
+   */
+  map(mapper) {
+    return new Observable((subscriber) => {
+      let idx = 0;
+      this.subscribe({
+        next(value) {
+          try {
+            subscriber.next(mapper(value, idx++));
+          } catch (err) {
+            subscriber.error(err);
+          }
+        },
+        error(error) {
+          subscriber.error(error);
+        },
+        complete() {
+          subscriber.complete();
+        },
+      }, {
+        signal: subscriber.signal,
+      });
+    });
+  }
+
+  /**
+   * @param {Predicate} predicate
+   * @returns {Observable}
+   */
+  filter(predicate) {
+    return new Observable((subscriber) => {
+      let idx = 0;
+      this.subscribe({
+        next(value) {
+          try {
+            if (predicate(value, idx++)) {
+              subscriber.next(value);
+            }
+          } catch (err) {
+            subscriber.error(err);
+          }
+        },
+        error(error) {
+          subscriber.error(error);
+        },
+        complete() {
+          subscriber.complete();
+        },
+      }, {
+        signal: subscriber.signal,
+      });
+    });
+  }
+
+  /**
+   * @param {number} amount
+   * @returns {Observable}
+   */
+  take(amount) {
+    return new Observable((subscriber) => {
```

Comment:
All the operators are missing argument validation and the checks should account for out of range values (i.e. what does `take(-1)` do? (currently it consumes everything)